lib/miq_expression/field.rb

```diff
@@ -1,15 +1,27 @@
 class MiqExpression::Field < MiqExpression::Target
-  REGEX = /
+  REGEX_BEFORE_DASH = /
 (?<model_name>([[:upper:]][[:alnum:]]*(::)?)+)
 (?!.*\b(managed|user_tag)\b)
 \.?(?<associations>[a-z][0-9a-z_\.]+)?
--
+/x
+
+  REGEX_AFTER_DASH = /
 (?:
   (?<virtual_custom_column>#{CustomAttributeMixin::CUSTOM_ATTRIBUTES_PREFIX}[a-z0-9A-Z]+[:_\-.\/[:alnum:]]*)|
   (?<column>[a-z]+(_[[:alnum:]]+)*)
 )
 /x
 
+  def self.match(field)
+    before, after = field.split("-", 2)
```

Comment:
You claim to not love this but the method is basically parsing models and attributes so it makes perfect sense to split on the hyphen.

Reply:
BTW, I did `split("-", 2)` on purpose also because the after part seems to allow hyphens in its value.  This ensure we only split on the first `-`

Reply:
Decided to _not_ split this, but we may choose to do that later. In particular the question of `-` being a value value before the `-` (which might happen in tags, but not fields). Easy enough to fix the perf issue in this PR without worrying about that, so deferring the split.

Reply:
Ok, it certainly make sense to only change the performance problem we care about and if that can be solved without changing this, 👍